server/core.go

```diff
@@ -0,0 +1,54 @@
+package server
+
+import (
+	logging "github.com/ipfs/go-log"
+	api "github.com/proximax-storage/go-xpx-dfms-api"
+
+	"github.com/valyala/fasthttp"
+)
+
+var log = logging.Logger("gateway")
+
+func init() {
+	logging.SetupLogging()
+}
+
+type gateway struct {
+	server  fasthttp.Server
+	address string
+}
+
+func NewGateway(api api.Client, opts ...GatewayOption) *gateway {
+	gopts := ParseOptions(opts...)
+
+	cfg, err := loadConfig(gopts.cfg)
+	if err != nil {
+		log.Fatal("Cannot load config: ", err)
+	}
+
+	gopts.ApplyToConfig(cfg)
+
+	if gopts.debug {
+		logging.SetDebugLogging()
```

Comment:
Use here: logging.SetLogLevel("DEBUG", "gateway"), otherwise you set debug to all the other modules in the system.